src/api.jl

```diff
@@ -255,16 +255,28 @@ end
 
 function get_engine(ctx::Context, engine::AbstractString; kw...)
     query = Dict("name" => engine, "deleted_on" => "")
-    rsp = _get(ctx, PATH_ENGINE; query = query, kw...).computes
-    length(rsp) == 0 && throw(HTTPError(404))
-    return rsp[1]
+    rsp = _get(ctx, PATH_ENGINE; query = query, kw...)
+    try
+        computes = rsp.computes
+        (computes === nothing || length(computes) == 0) && throw(HTTPError(404))
+        return computes[1]
+    catch
+        @warn "Caught exception when parsing JSON: $rsp"
+        rethrow()
+    end
```

Comment:
I don't see similar exception handling in the rest of the functions, should we add it here?

Reply:
working on it :pray:

Reply:
@geokollias done, can you give it another check ?

Reply:
@NRHelmi ah I meant probably we don't need to do any exception handling given we mostly rethrow the exception. What do you think?

Reply:
oh sorry :sweat_smile: then this PR has no added value 
closing this PR since the fix is originally pushed transaction service side https://github.com/RelationalAI/raicloud-control-plane/pull/4169
thanks @geokollias

Reply:
@geokollias and @NRHelmi:
The reason for this PR was that when there is an exception, we can _log the JSON_ we received.

In the situation seen previously, we got an exception: "No method matching `length(::Nothing)`", but it wasn't clear _why_ that exception was thrown.

So this PR was trying to ensure that we always log the JSON received if we hit an exception when parsing it. 🤷 up to you all if you want to proceed or not, but it seems useful to me